disnake/types/threads.py

```diff
@@ -23,12 +23,13 @@
 """
 
 from __future__ import annotations
-from typing import List, Literal, Optional, TypedDict
+from typing import List, Literal, Optional, TypedDict, Union
 
 from .snowflake import Snowflake
+from ..enums import ThreadArchiveDuration as ThreadArchiveDurationT
 
 ThreadType = Literal[10, 11, 12]
-ThreadArchiveDuration = Literal[60, 1440, 4320, 10080]
+ThreadArchiveDuration = Union[Literal[60, 1440, 4320, 10080], ThreadArchiveDurationT]
```

Comment:
```suggestion
from ..enums import ThreadArchiveDuration

ThreadType = Literal[10, 11, 12]
ThreadArchiveDurationT = Union[Literal[60, 1440, 4320, 10080], ThreadArchiveDuration]
```

Reply:
This means that anywhere that uses this typehint would need to be updated. However, this typehint isn't exposed in the public api, so this rename should be fine.